pkg/user/mock_tx_server_test.go

```diff
@@ -0,0 +1,126 @@
+package user_test
+
+import (
+	"context"
+	"net"
+	"testing"
+
+	"github.com/celestiaorg/celestia-app/v6/app/grpc/tx"
+	"github.com/celestiaorg/celestia-app/v6/pkg/user"
+	abci "github.com/cometbft/cometbft/abci/types"
+	"github.com/cometbft/cometbft/rpc/core"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdktx "github.com/cosmos/cosmos-sdk/types/tx"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/test/bufconn"
+)
+
+// mockTxServer implements both gRPC ServiceServer and TxServer interfaces for mocking broadcast and tx status responses
+type mockTxServer struct {
+	sdktx.UnimplementedServiceServer
+	tx.UnimplementedTxServer
+	txStatusResponses   map[string][]*tx.TxStatusResponse // txHash with sequence of responses
+	txStatusCallCounts  map[string]int                    // txHash with number of TxStatus calls made
+	broadcastCallCounts map[string]int                    // txHash with number of BroadcastTx calls made
```

Comment:
I don't dislike this, but aren't we're using other mocks for tests in this package?

we need to fix here or flup imo

Reply:
we can definitely delete the other one, but there should be only one for our own sanity
 
 totally fine to flup